# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/Notification.h
# Commit: 984db15bc9e2
# Full Hash: 984db15bc9e2b5baa20bdc86baaa1165c381d326
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 11: Migrate ShowPersistentNotification to use IPC r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227696
# ==============================================================================

diff -r 622a66cdac2e -r 984db15bc9e2 dom/notification/Notification.h
--- a/dom/notification/Notification.h	Thu Nov 07 17:32:07 2024 +0000
+++ b/dom/notification/Notification.h	Thu Nov 07 17:32:07 2024 +0000
@@ -311,12 +311,14 @@
   //
   // Note that aCx may not be in the compartment of aGlobal, but aOptions will
   // have its JS things in the compartment of aCx.
-  static already_AddRefed<Notification> CreateAndShow(
+  static already_AddRefed<Notification> Create(
       JSContext* aCx, nsIGlobalObject* aGlobal, const nsAString& aTitle,
       const NotificationOptions& aOptions, const nsAString& aScope,
       ErrorResult& aRv);
+  void ShowOnMainThread(ErrorResult& aRv);
 
   bool CreateActor(Promise* aPromise);
+  bool SendShow(Promise* aPromise);
 
   nsIPrincipal* GetPrincipal();
 